src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -76,15 +79,23 @@ public async Task StartAsync(CancellationToken cancellationToken = default)
             _logger.Started();
         }
 
-        private async Task HandleBackgroundException(BackgroundService backgroundService)
+        private async Task TryStartBackgroundServiceAsync(
+            BackgroundService backgroundService, CancellationToken cancellationToken)
         {
             try
             {
+                // Fire IHostedService.Start
+                await backgroundService.StartAsync(cancellationToken).ConfigureAwait(false);
                 await backgroundService.ExecuteTask.ConfigureAwait(false);
             }
             catch (Exception ex)
             {
                 _logger.BackgroundServiceFaulted(ex);
+                if (_options is { BackgroundServiceExceptionBehavior: BackgroundServiceExceptionBehavior.StopHost })
```

Comment:
I kind of think this is a little more convoluted than it needs to be. I know there are advantages to using pattern matching, but this is a simple enum check. I guess I would write it as:

```C#
if (_options.BackgroundServiceExceptionBehavior == BackgroundServiceExceptionBehavior.StopHost)
```

`_options` can never be null, so there is an unnecessary null check when using pattern matching here. It is also a lot easier to read (IMO) to write the "typical" if check.

Reply:
That makes sense, I'll update it - thanks